### DISCUSSION TITLE
Setting up mirrors for Homebrew bottles

### DISCUSSION THREAD

**Discussion by skyzh** [Homebrew/discussions#1917] [Getting started]
There have been many mirror sites that provide a copy of Homebrew bottles. Using those mirror sites could help you download Homebrew bottles faster in certain regions.

## Mirror List

| Operated By     | Usage Instructions                                          | Report Issues                                 |
|-----------------|-------------------------------------------------------------|-----------------------------------------------|
| TUNA          | https://mirrors.tuna.tsinghua.edu.cn/help/homebrew-bottles/ | https://github.com/tuna/issues                |
| BFSU (by TUNA) | https://mirrors.bfsu.edu.cn/help/homebrew-bottles/          | https://github.com/tuna/issues                |
| USTC            | https://mirrors.ustc.edu.cn/help/homebrew-bottles.html      | https://github.com/ustclug/discussions/issues |
| SJTUG           | https://mirrors.sjtug.sjtu.edu.cn/docs/homebrew-bottles     | https://github.com/sjtug/mirror-requests      |

## Notes

* Currently, we are using the almost-being-deprecated `HOMEBREW_BOTTLE_DOMAIN` for configuration.  And there is an ongoing effort for mirror sites (as of 2021-07-31) to migrate to new infrastructures and use the new `HOMEBREW_ARTIFACT_DOMAIN` for Homebrew mirror configurations. https://github.com/Homebrew/discussions/discussions/1906
* If you find brew not working as you expected after configuring mirror sites, and the bug is not related to brew itself, please report the issues to the mirror maintainers instead of reporting them in Homebrew discussions or issues. 

Feel free to update mirror-related information in this discussion!

**Comment by skyzh** [comment]
TUNA and BFSU mirrors are now working as expected.

**Comment by carlocab** [comment]
There is currently a bug where manifests aren't fetched from bottle mirrors. This results in issues like [this one](https://github.com/Homebrew/discussions/discussions/2530#discussioncomment-1999064).

Pull requests to fix this welcome. I think it is sufficient to teach `brew` to fetch manifests from bottle mirrors that are not ghcr.io.

**Comment by HomyeeKing** [comment]
the GNU gettext latest version is not available on homebrew 
0.21.1 on brew while latest is 0.22

**Reply by gromgit** [reply]
The `gettext 0.22` PR has hit some issues: https://github.com/Homebrew/homebrew-core/pull/134039. Perhaps you can help resolve them?

**Comment by Pix13** [comment]
Hi guys maybe not the best place, i'd like to mirror Homebrew on my infrastructure and offer it to the community ( i'm an official mirror of Rocky Linux, and the main host of RPMFusion), as i'm now using a Mac :) Is there a documentation for that ?

**Reply by SMillerDev** [reply]
Homebrew is mirrored by default since it uses GitHub infra for all hosting, there are therefore no guides for mirroring.
